deployment/scripts/upgrade.sh

```diff
@@ -0,0 +1,375 @@
+#! /bin/bash
+
+set -ex
+
+TIMES=300
+
+fluentd_nodes=""
+fluentd_nodeselector="logging-infra-fluentd=true"
+
+# this should be updated to get the lowest found version
```

Comment:
ITYM highest found version.

Also could you rename this, something like `getDeploymentVersion` ...? Something that hopefully doesn't confuse with the image/platform version, but indicates the structural version. Or `getIntegrationVersion` or `getStructuralVersion`?